src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs

```diff
@@ -8202,5 +8202,227 @@ .maxstack  2
 }
 ");
         }
+
+        [Fact]
+        public void ComplexTypes()
+        {
+            string originalSourceCode = @"
+using System;
+using System.Collections.Generic;
+
+class C
+{
+    public static void G()
+    {       
+        var <N:0>a</N:0> = new { key = ""a"", value = new List<(int, int)>()};
+        var <N:1>b</N:1> = (number: 5, value: a);
+        var <N:2>c</N:2> = new[] { b };
+        IntPtr[] <N:3>d</N:3> = new IntPtr[1];
```

Comment:
Thank you! This is done

---
In reply to: [116093049](https://github.com/dotnet/roslyn/pull/19283#discussion_r116093049) [](ancestors = 116093049)